### PR TITLE
I294 functions junit xml

### PR DESCRIPTION
**Pull request by ondrejtomas**
- Propagate error messages from CF call
- Add timestamps to logs
- Improve test_utils.py to return values

### CODE REVIEW

File: doc/WAW_functions_doc.md [hunk 1]
```diff
@@ -108,3 +115,19 @@ from evaluation.
     }
 ]
 ```
+
+**Output junit xml file example (see -j script parameter):**
+```
+<?xml version="1.0" encoding="utf-8"?>
+<testsuites errors="0" failures="0" tests="1" time="761.0">
+	<testsuite errors="0" failures="0" name="<TEST_FILE_NAME>" skipped="0" tests="1" time="761.0" timestamp="2019-05-24 12:20:22.173440">
+		<testcase name="test example 1" time="761">
```
Comment (Tereza-Pytelova) [inline doc/WAW_functions_doc.md]: To be consistent.. Or fill in the TEST_FILE_NAME
```suggestion
		<testcase name="<TEST_CASE_NAME>" time="761">
```

File: scripts/wawCommons.py [hunk 2]
```diff
@@ -555,10 +556,11 @@ def getFunctionResponseJson(cloudFunctionsUrl, urlNamespace, username, password,
...
         responseContentType = functionResponse.headers.get('content-type')
         if responseContentType != 'application/json':
-            logger.error('Response content type is not json, content type: %s, response:\n%s', responseContentType, functionResponse.text)
-            return None
+            errorMessage = "Response content type is not json, content type '{}', response:\n'{}'".format(responseContentType, functionResponse.text)
+            logger.error(errorMessage)
+            return None, errorMessage
 
-        return functionResponse.json()
+        return functionResponse.json(), None
```
Comment (Tereza-Pytelova) [inline scripts/wawCommons.py]: Isn't it better to use custom exception here?

File: ci/unit_tests/functions_test_evaluate/main_data/test_multi_junitxml_ref.eval.junit.xml [hunk 3]
```diff
@@ -26,5 +26,17 @@
...
+		</testcase>
+		<testcase name="Error from testing phase - only type">
+			<error type="error type"/>
+		</testcase>
+		<testcase name="Error from testing phase - message and type">
+			<error message="error message" type="error type"/>
+		</testcase>
+		<testcase name="Error from testing phase - message and type and rest of the test definition">
+			<error message="error message" type="error type"/>
+		</testcase>
```
Comment (Tereza-Pytelova) [inline ci/unit_tests/functions_test_evaluate/main_data/test_multi_junitxml_ref.eval.junit.xml:40]: I would expect the diff here ("test message" expected but "test message 2" returned)

Comment (ondrejtomas) [reply ci/unit_tests/functions_test_evaluate/main_data/test_multi_junitxml_ref.eval.junit.xml:40]: This is about propagation of error from testing phase, failure (where diff should be visible) could be found on line 23.

**Comment by ondrejtomas** [comment]
Even Nighly build passes without unexpected (deprecation) warnings:
https://travis-ci.com/IBM/watson-assistant-workbench/builds/113397696

File: scripts/wawCommons.py [hunk 4]
```diff
@@ -574,52 +584,65 @@ def getFunctionResponseJson(cloudFunctionsUrl, urlNamespace, username, password,
...
+                    None,
+                    {'activationId': activationId, 'contentType': responseContentType})
+                logger.error(exception)
+                raise exception
 
             responseJson = functionResponse.json()
             if isinstance(responseJson, dict) and 'result' in responseJson\
              and isinstance(responseJson['result'], dict) and 'payload' in responseJson['result']:
-                return responseJson['result']['payload']
+                return responseJson['result']['payload'], None
```
Comment (Tereza-Pytelova) [inline scripts/wawCommons.py]: remove "None"

**Review by Tereza-Pytelova — DISMISSED** [review]
Thank you for the effort!

**Review by Tereza-Pytelova — APPROVED** [review]

**Pull request merged**